client/scripts/Account.js

```diff
@@ -10,6 +11,76 @@ Account.prototype.login = function login (email, unhashedPass, callback) {
 	this.loginNoHash(email, CryptoJS.SHA256(unhashedPass).toString(CryptoJS.enc.Base64), callback);
 };
 
+Account.prototype.genericRequest = function (serverhook, parameters, callback) {
+	if (this.uKey !== ""){
+		var req = new XMLHttpRequest();
+
+		req.addEventListener("readystatechange", function (event) {
+			if (req.status == 200 && req.readyState == 4) {
+				var data = JSON.parse(req.responseText);
+
+				if (data.error) {
+					callback(data.error)						
+					return;
+				}
+				
+				setTimeout(callback, 0, null, data); // null = err param
+			} else if (req.readyState == 4) {
+				callback("Connection error, status code: " + req.status);
+			}
+		}.bind(this));
+		req.open("GET", this.server + "/" + serverhook + "?" + parameters);
+		req.send();
+	}
+	else {
+		callback("Not logged in!")
+	}
+};
+
+Account.prototype.setCoordFavorite = function (newX, newY, x, y, name, callback) {
+	var parameters = "uKey=" + encodeURIComponent(this.uKey) 
+								 + "&room=" + encodeURIComponent(room)
+								 + "&newX=" + encodeURIComponent(newX)
+								 + "&newY=" + encodeURIComponent(newY)
+								 + "&x=" + encodeURIComponent(x)
+								 + "&y=" + encodeURIComponent(y)
+								 + "&name=" + encodeURIComponent(name);
+	this.genericRequest("setcoordfavorite", parameters, callback);
```

Comment:
This can be changed to

```javascript
this.request("/hook", {
    rom: room,
    newX: newX,
    newY: newY,
    x: x,
    y: y,
    name: name
}, this.parseData.bind(this, callback));
```